### DISCUSSION TITLE
Lower records

### DISCUSSION THREAD

**Discussion by JonathanStarup** [flix/flix#8490] [Ideas]
It seems like we could implement records in flix and just lower to that, a la datalog or channels

**Comment by magnus-madsen** [comment]
What do you mean?

**Reply by JonathanStarup** [reply]
records are just a tagged linked list, so we could implement it in flix and lower to that, instead of the bytecode implementation

**Reply by magnus-madsen** [reply]
Does that give O(1) access? Structs are mostly intended to be used for data structures where performance matters.

**Reply by JonathanStarup** [reply]
It doesn't change anything, its just to have the implementation in flix instead of bytecode - easier to read/write/maintain

**Reply by magnus-madsen** [reply]
What would you lower to? Tuples?

**Reply by JonathanStarup** [reply]
enums/structs/tuples yeah
